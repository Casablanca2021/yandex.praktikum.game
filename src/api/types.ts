export type SignInData = {
  login: string;
  password: string;
};

export type SignUpResponse = {
  id: number;
};

export type SignInResponse = string;

export type UserInfoResponse = {
  id: number;
  // eslint-disable-next-line camelcase
  first_name: string;
  // eslint-disable-next-line camelcase
  second_name: string;
  // eslint-disable-next-line camelcase
  display_name: string;
  login: string;
  email: string;
  phone: string;
  avatar: string;
};

export type SignUpData = Omit<UserInfoResponse, 'display_name' | 'avatar' | 'id'>

export type LogOutResponse = string;
