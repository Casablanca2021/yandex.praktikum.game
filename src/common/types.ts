export type InputChangeEvent = React.ChangeEvent<HTMLInputElement>;

export type Nullable<T> = undefined extends T ? void : T;
