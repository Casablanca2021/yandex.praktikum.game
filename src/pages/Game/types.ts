import logo from 'assets/Road.png';
import car from 'assets/car.png';
import { animate } from 'utils';

export class Road {
  private y = 0;

  private speed = 30;

  private height: number;

  private width: number;

  private image = new Image();

  private ctx: CanvasRenderingContext2D | null | undefined;

  constructor(width: number, height: number) {
    this.height = height;
    this.width = width;
  }

  init(ctx: CanvasRenderingContext2D): void {
    this.image.onload = () => {
      this.ctx = ctx;
      this.draw();
    }
    this.image.src = logo;
  }

  draw(): void {
    this.y += this.speed;

    if (this.y > this.height) { this.y = this.speed; }

    this.ctx?.clearRect(0, 0, this.width, this.height);

    //Цикл для отрисовки дороги 
    for (let i = 0; i < 2; i += 1) {
      this.ctx?.drawImage(
        this.image,
        0,
        0,
        this.image.width,
        this.image.height,
        0,
        this.y - (this.height * i),
        this.width,
        this.height,
      );
    }
  }
}

export class Car {

  private height = 0;
  private width = 0;
  private carHeight = 150;
  private carWidth = 100;
  private y = 0;
  private x = 0;
  private maxHeight = 0;
  private maxWidth = 0;
  private image = new Image();
  private ctx: CanvasRenderingContext2D | null | undefined;

  constructor(width: number, height: number) {
    this.height = height;
    this.width = width;
    this.maxHeight = this.height - this.carHeight;
    this.maxWidth = this.width - this.carWidth;
    window.addEventListener('keydown', (e) => {
      if (e.key === 'ArrowRight') {
        this.toRight();
      }
      if (e.key === 'ArrowLeft') {
        this.toLeft();
      }
      if (e.key === 'ArrowUp') {
        this.toUp();
      }
      if (e.key === 'ArrowDown') {
        this.toDown();
      }
    });
  }

  private setX = (value: number) => {
    let x = this.x + value;
    if (x < 0) {
      x = 0;
    }

    if (x > this.width - this.carWidth) {
      x = this.maxWidth;
    }

    this.x = x;
  }

  private setY = (value: number) => {
    let y = this.y + value;
    if (y < 0) {
      y = 0;
    }

    if (y > this.height - this.carHeight) {
      y = this.maxHeight;
    }

    this.y = y;
  }

  toLeft() {
    animate((time: number) => time + 5, this.setX, 300, -1)
  }

  toRight() {
    animate((time: number) => time + 5, this.setX, 300, 1)
  }

  toUp() {
    animate((time: number) => time + 5, this.setY, 300, -1)
  }

  toDown() {
    animate((time: number) => time + 5, this.setY, 300, 1)
  }

  init(ctx: CanvasRenderingContext2D): void {
    this.x = this.width/2;
    this.y = this.height;
    this.image.onload = () => {
      this.ctx = ctx;
      this.draw();
    }
    this.image.src = car;
  }

  draw(): void {
    this.ctx?.clearRect(0, 0, this.width, this.height);
    this.ctx?.drawImage(
      this.image,
      0,
      0,
      this.image.width,
      this.image.height,
      this.x,
      this.y,
      this.carWidth,
      this.carHeight,
    );
  }
}
