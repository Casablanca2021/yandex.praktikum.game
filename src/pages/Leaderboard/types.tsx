type Leaders = {
    id: number;
    name: string;
    points: number;
    avatar: string;
}

type LeaderboardPage = {
    activeItem: string;
    leaders: Leaders[];
}

export { LeaderboardPage, Leaders };
