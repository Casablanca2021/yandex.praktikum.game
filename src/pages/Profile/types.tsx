type Profile = {
    id: number,
    first_name: string;
    second_name: string;
    display_name: string;
    login: string;
    email: string;
    phone: string;
    avatar: string;
}

type profileValidations = Record<keyof Profile, boolean>

type ProfileForm = {
    profile :Partial<Profile>;
    isError: Partial<profileValidations>;
}

export default ProfileForm;
