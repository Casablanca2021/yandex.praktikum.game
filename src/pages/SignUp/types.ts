export type Fields = {
  first_name: string;
  second_name: string;
  login: string;
  email: string;
  phone: string;
  password: string;
  password_confirm: string;
};

export type FieldErrors = Record<keyof Fields, boolean>;
