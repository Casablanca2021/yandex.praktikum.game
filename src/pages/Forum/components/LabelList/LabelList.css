.label-list {
    margin-left: -2px;
    margin-right: -2px;
    margin-top: 4px;
}

.label-list__item {
    display: inline-block;
    padding: 4px 8px;
    background-color: #000;
    margin: 2px;
    color: #ffffff;
    border-radius: 5px;
    cursor: pointer;
}

.label-list__item_active {
    background-color: #ffffff;
    color: #000000;
}
