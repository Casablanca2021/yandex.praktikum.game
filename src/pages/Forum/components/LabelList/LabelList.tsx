import React, { FC } from 'react';
import { OwnProps } from 'pages/Forum/components/LabelList/types';
import './LabelList.css';
import { Link } from 'react-router-dom';

const LabelList: FC<OwnProps> = (props) => {
  return (
    <div className="label-list">
      {props.items.map((item, key) => {
        return <Link to={item.path} key = {key} className={`label-list__item ${item.active ? 'label-list__item_active' : ''}`}>{item.name}</Link>
      })}
    </div>
  );
};

export default LabelList;