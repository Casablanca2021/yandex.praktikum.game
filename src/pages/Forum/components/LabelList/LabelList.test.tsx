import * as React from 'react';
import * as renderer from 'react-test-renderer';
import LabelList from 'pages/Forum/components/LabelList/index';
import { Items } from 'pages/Forum/components/LabelList/types';
import { Provider } from 'react-redux';
import store from 'store';
import history from 'store/history';
import { ConnectedRouter } from 'connected-react-router';

describe(`LabelList Component`, () => {
  it(`renders correctly with some items`, () => {
    const items: Items = [
      {
        name: 'Test 1',
        path: '/test-1',
      },
      {
        name: 'Test 2',
        path: '/test-2',
      },
    ];

    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <LabelList items={items} />
          </ConnectedRouter>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders correctly with some other items`, () => {
    const items: Items = [
      {
        name: 'Test 1',
        path: '/test-1',
      },
      {
        name: 'Test 2',
        path: '/test-2',
      },
      {
        name: 'Test 3',
        path: '/test-3',
      },
    ];

    const tree = renderer
      .create(<Provider store={store}>
        <ConnectedRouter history={history}>
          <LabelList items={items} />
        </ConnectedRouter>
      </Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
