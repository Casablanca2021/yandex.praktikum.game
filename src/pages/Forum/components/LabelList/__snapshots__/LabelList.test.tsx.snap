// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelList Component renders correctly with some items 1`] = `
<div
  className="label-list"
>
  <a
    className="label-list__item "
    href="/test-1"
    onClick={[Function]}
  >
    Test 1
  </a>
  <a
    className="label-list__item "
    href="/test-2"
    onClick={[Function]}
  >
    Test 2
  </a>
</div>
`;

exports[`LabelList Component renders correctly with some other items 1`] = `
<div
  className="label-list"
>
  <a
    className="label-list__item "
    href="/test-1"
    onClick={[Function]}
  >
    Test 1
  </a>
  <a
    className="label-list__item "
    href="/test-2"
    onClick={[Function]}
  >
    Test 2
  </a>
  <a
    className="label-list__item "
    href="/test-3"
    onClick={[Function]}
  >
    Test 3
  </a>
</div>
`;
